src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -1,69 +1,83 @@
         {
-            var expression = VisitExpression(node.Expression);
-            var result = LowerPattern(node.Pattern, expression);
-            return result;
+            var loweredExpression = VisitExpression(node.Expression);
+            var loweredPattern = LowerPattern(node.Pattern);
+            return MakeIsPattern(loweredPattern, loweredExpression);
         }
 
         // Input must be used no more than once in the result. If it is needed repeatedly store its value in a temp and use the temp.
-        BoundExpression LowerPattern(BoundPattern pattern, BoundExpression input)
+        BoundExpression MakeIsPattern(BoundPattern loweredPattern, BoundExpression loweredInput)
         {
-            var syntax = _factory.Syntax = pattern.Syntax;
-            switch (pattern.Kind)
+            var syntax = _factory.Syntax = loweredPattern.Syntax;
+            switch (loweredPattern.Kind)
             {
                 case BoundKind.DeclarationPattern:
                     {
-                        var declPattern = (BoundDeclarationPattern)pattern;
-                        return LowerDeclarationPattern(declPattern, input);
+                        var declPattern = (BoundDeclarationPattern)loweredPattern;
+                        return MakeIsDeclarationPattern(declPattern, loweredInput);
                     }
 
                 case BoundKind.WildcardPattern:
                     return _factory.Literal(true);
 
                 case BoundKind.ConstantPattern:
                     {
-                        var constantPattern = (BoundConstantPattern)pattern;
-                        return LowerConstantPattern(constantPattern, input);
+                        var constantPattern = (BoundConstantPattern)loweredPattern;
+                        return MakeIsConstantPattern(constantPattern, loweredInput);
                     }
 
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(pattern.Kind);
+                    throw ExceptionUtilities.UnexpectedValue(loweredPattern.Kind);
+            }
+        }
+
+        BoundPattern LowerPattern(BoundPattern pattern)
+        {
+            switch (pattern.Kind)
+            {
+                case BoundKind.DeclarationPattern:
+                    {
+                        var declPattern = (BoundDeclarationPattern)pattern;
+                        return declPattern.Update(declPattern.Variable, VisitExpression(declPattern.VariableAccess), declPattern.DeclaredType, declPattern.IsVar);
+                    }
+                case BoundKind.ConstantPattern:
+                    {
+                        var constantPattern = (BoundConstantPattern)pattern;
+                        return constantPattern.Update(VisitExpression(constantPattern.Value), constantPattern.ConstantValue);
+                    }
+                default:
+                    return pattern;
```

Comment:
I will add a case for wildcard, and assert that the switch is complete in a subsequent PR.